File: fedml/src/main/java/nl/tudelft/trustchain/fedml/ai/dataset/har/HARIterator.java
package nl.tudelft.trustchain.fedml.ai.dataset.har;

import org.deeplearning4j.datasets.fetchers.DataSetType;
import org.nd4j.linalg.dataset.api.iterator.BaseDatasetIterator;

import java.io.File;
import java.io.IOException;

import nl.tudelft.trustchain.fedml.ai.IteratorDistributions;
import nl.tudelft.trustchain.fedml.ai.MaxTestSamples;

public class HARIterator extends BaseDatasetIterator {

    /**
     * Constructor to get the full HAL data set (either test or train sets) with shuffling based on a random seed.
     */
    public HARIterator(File baseDirectory, int batchSize, int seed, IteratorDistributions iteratorDistribution, DataSetType dataSetType) throws IOException {
        super(batchSize, -1, new HARDataFetcher(baseDirectory, seed, iteratorDistribution.getValue(), dataSetType, null));
    }

    /**
     * Constructor to get the full HAL data set (either test or train sets) with shuffling based on a random seed.
     */
    public HARIterator(File baseDirectory, int batchSize, int seed, IteratorDistributions iteratorDistribution, DataSetType dataSetType, MaxTestSamples maxTestSamples) throws IOException {
        super(batchSize, -1, new HARDataFetcher(baseDirectory, seed, iteratorDistribution.getValue(), dataSetType, maxTestSamples));
    }
}

File: fedml/src/main/java/nl/tudelft/trustchain/fedml/ai/dataset/har/HARIterator.kt
package nl.tudelft.trustchain.fedml.ai.dataset.har

import nl.tudelft.trustchain.fedml.Behaviors
import nl.tudelft.trustchain.fedml.ai.DatasetIteratorConfiguration
import org.deeplearning4j.datasets.fetchers.DataSetType
import org.nd4j.linalg.dataset.api.iterator.BaseDatasetIterator
import java.io.File


class HARIterator(
    iteratorConfiguration: DatasetIteratorConfiguration,
    seed: Int,
    dataSetType: DataSetType,
    baseDirectory: File,
    behavior: Behaviors
) : BaseDatasetIterator(
    iteratorConfiguration.batchSize.value,
    -1,
    HARDataFetcher(
        baseDirectory,
        seed,
        iteratorConfiguration.distribution.value,
        dataSetType,
        iteratorConfiguration.maxTestSamples?.value ?: Integer.MAX_VALUE
    )
)
